cdap-integration-test/src/test/java/co/cask/cdap/test/IntegrationTestBaseTest.java

```diff
@@ -40,4 +49,26 @@ public void testFlowManager() throws Exception {
     flowManager.stop();
   }
 
+  @Test
+  public void testDeployApplicationInNamespace() throws Exception {
+    Id.Namespace namespace = new Id.Namespace ("Test1");
+    createNamespace(new NamespaceMeta.Builder().setName(namespace).build());
+    ClientConfig clientConfig = new ClientConfig.Builder(getClientConfig()).build();
+    clientConfig.setNamespace(namespace);
```

Comment:
Seems strange to me to allow a `setNamespace` method if you have the `createNamespacedClientConfig` method above. Shouldn't `ClientConfig` be an immutable class @anwar6953 ?